### PR TITLE
Refactor farm

### PR DESCRIPTION
**Pull request by linkielink**
refactored farm by merging vault and astroLp components

### CODE REVIEW

File: src/components/Modals/AddFarmAssets/index.tsx [hunk 1]
```diff
@@ -9,29 +9,20 @@ import useStore from 'store'
...
+    useStore.setState({
+      addFarmBorrowingsModal: null,
+      farmModal: farmModal ? { ...farmModal, selectedBorrowDenoms: selectedDenoms } : null,
+    })
   }
 
   const updateSelectedDenoms = useCallback((denoms: string[]) => setSelectedDenoms(denoms), [])
 
-  const showContent = modal && !!farm
+  const showContent = modal
```
Comment (bobthebuidlr) [inline src/components/Modals/AddFarmAssets/index.tsx:25]: Unnecessary to create this variable

File: src/components/account/AccountStrategiesTable/functions.ts [hunk 2]
```diff
@@ -49,11 +49,11 @@ export function getVaultAccountStrategiesRow(
     coinsChange: {
       primary: BNCoin.fromDenomAndBigNumber(
         vault.denoms.primary,
-        !prev ? BN_ZERO : primaryAmount.minus(primaryAmountPrev),
+        !prev ? primaryAmount : primaryAmount.minus(primaryAmountPrev),
```
Comment (bobthebuidlr) [inline src/components/account/AccountStrategiesTable/functions.ts:52]: WHy did you add this to the change? If there is no previous, there is no change, so `BN_ZERO` seems correct to me

File: src/components/account/AccountStrategiesTable/functions.ts [hunk 3]
```diff
@@ -49,11 +49,11 @@ export function getVaultAccountStrategiesRow(
...
       ),
       secondary: BNCoin.fromDenomAndBigNumber(
         vault.denoms.secondary,
-        !prev ? BN_ZERO : secondaryAmount.minus(secondaryAmountPrev),
+        !prev ? secondaryAmount : secondaryAmount.minus(secondaryAmountPrev),
```
Comment (bobthebuidlr) [inline src/components/account/AccountStrategiesTable/functions.ts:56]: Same here of course

File: src/components/account/AccountStrategiesTable/functions.ts [hunk 4]
```diff
@@ -146,11 +146,11 @@ export function getAstroLpAccountStrategiesRow(
     coinsChange: {
       primary: BNCoin.fromDenomAndBigNumber(
         astroLp.denoms.primary,
-        !prev ? BN_ZERO : primaryAmount.minus(primaryAmountPrev),
+        !prev ? primaryAmount : primaryAmount.minus(primaryAmountPrev),
```
Comment (bobthebuidlr) [inline src/components/account/AccountStrategiesTable/functions.ts:149]: And on these two lines

File: src/hooks/accounts/useUpdatedAccount/functions.ts [hunk 5]
```diff
@@ -70,15 +70,35 @@ export function addValueToVaults(
...
       vaults[index].values.secondary = BN(vaults[index].values.secondary).plus(halfValue)
     } else {
-      const vault = vaults.find((vault) => vault.address)
+      const vault = availableVaults.find((vault) => vaultValue.address === vault.address)
 
       if (!vault) return
       const apy = availableVaults.find((vault) => vault.address === vaultValue.address)?.apy ?? null
       const apr = availableVaults.find((vault) => vault.address === vaultValue.address)?.apr ?? null
 
+      console.log(vaultValues[0].address, availableVaults)
```
Comment (bobthebuidlr) [inline src/hooks/accounts/useUpdatedAccount/functions.ts:79]: log

File: src/hooks/accounts/useUpdatedAccount/functions.ts [hunk 6]
```diff
@@ -70,15 +70,35 @@ export function addValueToVaults(
...
+
       vaults.push({
-        ...vault,
-        ...MOCK_DEPOSITED_VAULT_POSITION,
+        status: 'active',
```
Comment (bobthebuidlr) [inline src/hooks/accounts/useUpdatedAccount/functions.ts:82]: Why don't you spread the current vault anymore to get the metadata? What changed ?

**Review by bobthebuidlr — CHANGES_REQUESTED** [review]

File: src/components/account/AccountStrategiesTable/functions.ts (hunk 2 above)
Comment (linkielink) [reply src/components/account/AccountStrategiesTable/functions.ts:52]: If there is no previous, the position is new and should be green. Like it is for depositing and lending as well.

File: src/hooks/accounts/useUpdatedAccount/functions.ts (hunk 6 above)
Comment (linkielink) [reply src/hooks/accounts/useUpdatedAccount/functions.ts]: This was a type debugging thing, I changed it back.

Comment (linkielink) [reply src/hooks/accounts/useUpdatedAccount/functions.ts]: This was a type debugging thing, I changed it back.

**Review by bobthebuidlr — APPROVED** [review]

**Pull request merged**